### PR TITLE
New pull request for Issue #1

### PR DESCRIPTION
**Pull request by aljana**
New request for issue #1. Unrelated commits are now removed. Hope it's ok.

### CODE REVIEW

File: piplmesh/urls.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from django.conf.urls.defaults import patterns, include, url
+from frontend.views import *
+
+# Uncomment the next two lines to enable the admin:
+# from django.contrib import admin
+# admin.autodiscover()
+
+urlpatterns = patterns('',
+	('^$', first),
```
Comment (mitar) [inline piplmesh/urls.py]: Please align.

Comment (mitar) [inline piplmesh/urls.py]: Name it maybe "home"?

File: piplmesh/frontend/views.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Create your views here.
+
+from django.template import RequestContext
+from django.http import HttpResponse
+from django.shortcuts import render_to_response
+
+def first(request):
```
Comment (mitar) [inline piplmesh/frontend/views.py]: Please use [class-base generic veiws](https://docs.djangoproject.com/en/dev/ref/class-based-views/).

File: piplmesh/settings.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    'django.contrib.messages.middleware.MessageMiddleware',
+)
+
+ROOT_URLCONF = 'piplmesh.urls'
+
+TEMPLATE_DIRS = (
+    # Put strings here, like "/home/html/django_templates" or "C:/www/django/templates".
+    # Always use forward slashes, even on Windows.
+    # Don't forget to use absolute paths, not relative paths.
+	os.path.join(os.path.dirname(__file__), 'frontend/templates').replace('\\','/'),
```
Comment (mitar) [inline piplmesh/settings.py]: No, this is not OK. We are using Django app template discovery.

File: piplmesh/settings.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.sites',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    # Uncomment the next line to enable the admin:
+    # 'django.contrib.admin',
+    # Uncomment the next line to enable admin documentation:
+    # 'django.contrib.admindocs',
+    'frontend',
```
Comment (mitar) [inline piplmesh/settings.py]: It should be `piplmesh.frontend`.

File: piplmesh/urls.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    # Examples:
+    # url(r'^$', 'piplmesh.views.home', name='home'),
+    # url(r'^piplmesh/', include('piplmesh.foo.urls')),
+
+    # Uncomment the admin/doc line below to enable admin documentation:
+    # url(r'^admin/doc/', include('django.contrib.admindocs.urls')),
+
+    # Uncomment the next line to enable the admin:
+    # url(r'^admin/', include(admin.site.urls)),
+
```
Comment (mitar) [inline piplmesh/urls.py]: No unnecessary empty lines please.

File: piplmesh/settings.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+# List of finder classes that know how to find static files in
+# various locations.
+STATICFILES_FINDERS = (
+    'django.contrib.staticfiles.finders.FileSystemFinder',
+    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
+#    'django.contrib.staticfiles.finders.DefaultStorageFinder',
+)
+
+# Make this unique, and don't share it with anybody.
+SECRET_KEY = '02dl2nfiacp)87-1g2$=l@b(q5+qs^)qo=byzdvgx+35q)gw&^'
```
Comment (mitar) [inline piplmesh/settings.py]: This obviously is not the good way to do it. You should not be committing this to the repository. Check how we are doing it in [nodewatcher](http://dev.wlan-si.net/browser/nodewatcher).

File: piplmesh/settings.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+ADMIN_MEDIA_PREFIX = '/media/'
+
+# Additional locations of static files
+#STATICFILES_DIRS = (
+    # Put strings here, like "/home/html/static" or "C:/www/django/static".
+    # Always use forward slashes, even on Windows.
+    # Don't forget to use absolute paths, not relative paths.
+    
+#)
+
```
Comment (Irrialite) [inline piplmesh/settings.py:82]: I'm not sure what this is for and if commenting it was the right thing to do. Same with finders below this line. It doesn't exist in nodewatcher.

Comment (mitar) [inline piplmesh/settings.py:82]: Stable version of nodewatcher uses an older Django version (1.2) which has not yet had support for [static files](https://docs.djangoproject.com/en/dev/howto/static-files/). You should uncomment those back (and remove empty lines).

File: piplmesh/settings.py [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+# List of finder classes that know how to find static files in
+# various locations.
+#STATICFILES_FINDERS = (
+#    'django.contrib.staticfiles.finders.FileSystemFinder',
+#    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
+#    'django.contrib.staticfiles.finders.DefaultStorageFinder',
+#)
+
+# Make this unique, and don't share it with anybody.
+SECRET_KEY = '02dl2nfiacp)87-1g2$=l@b(q5+qs^)qo=byzdvgx+35q)gw&^'
```
Comment (mitar) [inline piplmesh/settings.py]: Still public!

Comment (aljana) [inline piplmesh/settings.py]: Here is one of the possible solutions (if anyone has time to implement it): http://stackoverflow.com/questions/4664724/distributing-django-projects-with-unique-secret-keys

**Comment by aljana** [comment]
Hmm, it would be nice to know if this is okay now or still needs improvements.

**Pull request merged**